tests/conftest.py

```diff
@@ -39,6 +47,26 @@ def start():
     APP.run("127.0.0.1", 12345)
 
 
+def wait_for_webserver_connection():
+    """Wait for the webserver to respond to HTTP requests."""
+    LOGGER.info("Testing for webserver connectivity.")
+    query_handler = GraphQLQueryHandler("http://127.0.0.1:12345/graphql")
+    timeout = time.time() + 30
+    while time.time() < timeout:
+        try:
+            query_handler.execute("{nothing}")
+            LOGGER.info("Up and running")
+            return
+        except HTTPError:  # BadRequest means the webserver came up.
+            LOGGER.info("Up and running")
+            return
+        except Exception as exception:  # pylint: disable=broad-except
+            LOGGER.error("%r", exception)
+        LOGGER.warning("Connection could not be established. Trying again in 1s..")
+        time.sleep(1)
+    raise TimeoutError("Timeout waiting for the webserver to start.")
```

Comment:
Why not use the [retry package](https://pypi.org/project/retry/) instead?

```python
@retry.retry(exceptions=Exception, tries=30, delay=1, logger=LOGGER)
def wait_for_webserver_connection():
    query_handler = GraphQLQueryHandler("http://127.0.0.1:12345/graphql")
    try:
        query_handler.execute("{nothing}")
    except HTTPError as err:
        if err.response.status_code != 400:
            raise
```

It won't raise TimeoutError if the startup fails though.

Reply:
Since this is the only place in the entire code-base where we do this kind of retry, I would say that the additional "overhead" of installing the retry package feels unnecessary.

If we were to add more retries, then definitely.